assets/src/edit-story/components/canvas/karma/multiSelectionMovable.karma.js

```diff
@@ -0,0 +1,172 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { Fixture } from '../../../karma';
+import { useStory } from '../../../app/story';
+import useInsertElement from '../useInsertElement';
+import { TEXT_ELEMENT_DEFAULT_FONT } from '../../../app/font/defaultFonts';
+
+describe('Multi-selection Movable integration', () => {
+  let fixture;
+
+  beforeEach(async () => {
+    fixture = new Fixture();
+
+    await fixture.render();
+  });
+
+  afterEach(() => {
+    fixture.restore();
+  });
+
+  describe('multi-selection', () => {
+    let element1;
+    let element2;
+    let element3;
+    let frame1;
+    let frame2;
+    let frame3;
+
+    beforeEach(async () => {
+      const insertElement = await fixture.renderHook(() => useInsertElement());
+      element1 = await fixture.act(() =>
+        insertElement('text', {
+          font: TEXT_ELEMENT_DEFAULT_FONT,
+          content: 'Text A',
+          x: 10,
+          y: 10,
+          width: 50,
+          height: 50,
+        })
+      );
+
+      element2 = await fixture.act(() =>
+        insertElement('text', {
+          font: TEXT_ELEMENT_DEFAULT_FONT,
+          content: 'Text B',
+          x: 100,
+          y: 100,
+          width: 50,
+          height: 50,
+        })
+      );
+
+      element3 = await fixture.act(() =>
+        insertElement('text', {
+          font: TEXT_ELEMENT_DEFAULT_FONT,
+          content: 'Text B',
+          x: element1.x + element1.width + 1,
+          y: element1.y + element1.height + 1,
+          width: 20,
+          height: 20,
+        })
+      );
+
+      frame1 = getElementFrame(fixture, element1.id);
+      frame2 = getElementFrame(fixture, element2.id);
+      frame3 = getElementFrame(fixture, element3.id);
+    });
+
+    it('should render initial content', () => {
+      expect(frame1.textContent).toEqual('Text A');
+      expect(frame2.textContent).toEqual('Text B');
+    });
+
+    describe('click interaction', () => {
+      let safezone;
+      beforeEach(async () => {
+        await clickOnTarget(fixture, frame1);
+        await clickOnTarget(fixture, frame2, 'Shift');
+
+        safezone = fixture.querySelector('[data-testid="safezone"]');
+      });
+
+      it('should have multiple elements selected', async () => {
+        const storyContext = await fixture.renderHook(() => useStory());
+        expect(storyContext.state.selectedElementIds).toEqual([
+          element1.id,
+          element2.id,
+        ]);
```

Comment:
In general, would it not be better to test that selection was actually these elements, rather than "just" verify that story provider says it is these elements? Like, verify that single or multi select movable includes those element frames?

I'm not sure myself here - we could have another test verify, that if story provider says these are to be selected, movable always contains the right ones and thus not worry about it here, but I feel like this is cheating a bit.

Reply:
> we could have another test verify, that if story provider says these are to be selected, movable always contains the right ones and thus not worry about it here

I think that's a good option (test the login form once and then just set the auth data to speed up the tests).
Or we can just create a utility function/matcher to check the selection and if we ever find a case where it fails (element is selected in the reducer but there is a UI issue) - we can switch to another implementation.